torchrl/data/replay_buffers/storages.py

```diff
@@ -38,11 +41,28 @@ def set(self, cursor: int, data: Any):
     def get(self, index: int) -> Any:
         raise NotImplementedError
 
+    def attach(self, buffer: Any) -> None:
+        """This function attaches a buffer to this storage.
+
+        Replay Buffers that read from this storage must call this
+        method to attach themselves. This guarantees that when data
+        in the storage changes, all relevant pieces of the buffer are
+        aware of it even if the storage is shared with another buffer
+        (eg. Priority Samplers).
+
+        Args:
+            buffer: the object that reads from this storage.
+        """
+        self.attached_entities.add(buffer)
+
     def __getitem__(self, item):
         return self.get(item)
 
     def __setitem__(self, index, value):
-        return self.set(index, value)
+        ret = self.set(index, value)
+        for i in self.attached_entities:
```

Comment:
usually (at least in our code base) `i` points to an integer.
Can we rename this?

Reply:
sure.